### PR TITLE
[dry-actions] 🧶 Abstract generic fetchObject(s) actions.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
The goal is to create reusable actions for most object types. We could probably add a generic `createObject` action too, and `deleteObject` if we need it.

drying dog

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
> I think this looks good and clean. Idk how I feel about the name being `object(s)` but if generic is what we are going for, that is it.

Yeah, I'm open to other suggestions. I'd like these actions to be used for various things (e.g. products, projects, tasks, etc) -- and "object" was the best I could come up with. 🤷‍♂

> One question I have is what does the new `reset` boolean flag do?

Again, I'm open to other suggestions for structure or names. It flags the difference between "I'm fetching a list of things, and the results should override whatever we already know about" vs "I'm fetching a list of things (e.g. page 2), and the results should be added to whatever we already know about".

File: src/js/store/products/reducer.ts [hunk 1]
```diff
@@ -24,39 +26,51 @@ const defaultState = {
 
 const reducer = (
   products: ProductsState = defaultState,
-  action: ProductsAction | LogoutAction,
+  action: ProductsAction | ObjectsAction | LogoutAction,
```
Comment (wlonk) [inline src/js/store/products/reducer.ts]: Still using `ProductsAction` here?

Comment (jgerigmeyer) [reply src/js/store/products/reducer.ts]: There is still one custom `ProductsAction`, for syncing the repos.

**Comment by davisagli** [comment]
Nice!

**Pull request merged**